rustls/src/quic.rs

```diff
@@ -371,15 +373,26 @@ pub trait ServerQuicExt {
             config.max_early_data_size == 0 || config.max_early_data_size == 0xffff_ffff,
             "QUIC sessions must set a max early data of 0 or 2^32-1"
         );
-        let mut imp =
-            ServerSessionImpl::new(config, vec![ServerExtension::TransportParameters(params)]);
+        let ext = match quic_version {
+            Version::Draft1 => ServerExtension::TransportParametersDraft(params),
+            Version::V1 => ServerExtension::TransportParameters(params),
+        };
+        let mut imp = ServerSessionImpl::new(config, vec![ext]);
         imp.common.protocol = Protocol::Quic;
         ServerSession { imp }
     }
 }
 
 impl ServerQuicExt for ServerSession {}
 
+#[non_exhaustive]
+pub enum Version {
+    /// Draft versions prior to V1
+    Draft1,
```

Comment:
Sorry, final nit: `Draft1` reads a bit too much like draft-1, maybe make it `V1Draft` or something similar?

Reply:
Oh, yeah, that's way better!